rtengine/filmnegativethumb.cc

```diff
@@ -35,6 +35,39 @@ extern const Settings* settings;
 
 }
 
+namespace
+{
+
+using rtengine::Imagefloat;
+using rtengine::findMinMaxPercentile;
+
+
+void calcMedians(
+    const Imagefloat* baseImg,
+    const int x1, const int y1,
+    const int x2, const int y2,
+    float &rmed, float &gmed, float &bmed
+)
+{
+    // Channel vectors to calculate medians
+    std::vector<float> rv, gv, bv;
```

Comment:
You could [reserve](https://en.cppreference.com/w/cpp/container/vector/reserve) `std::max(0, (y2 - y1) * (x2 - x1))` elements each to prevent reallocation.